mmv1/third_party/terraform/data_sources/data_source_vpc_access_connector.go

```diff
@@ -0,0 +1,31 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceVPCAccessConnector() *schema.Resource {
+
+	dsSchema := datasourceSchemaFromResourceSchema(resourceVPCAccessConnector().Schema)
+	addRequiredFieldsToSchema(dsSchema, "name")
+
+	return &schema.Resource{
+		Read:   dataSourceVPCAccessConnectorRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceVPCAccessConnectorRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	id, err := replaceVars(d, config, "projects/{{project}}/locations/{{region}}/connectors/{{name}}")
```

Comment:
Oh, sorry, didn't notice this on my first pass- we'll want to introduce `project` and `region` as optional fields as well. As-is, the datasource can only reference resources that are present in the user's default project + zone.

There's no need to test that that specifying the resource that way works explicitly, it's done through common code (`replaceVars`). We'll need to document the optional fields, though.

Reply:
Hey, no problem! I think I misunderstood `datasourceSchemaFromResourceSchema` and thought it copied `project` and `region` but it looks like even though the parameters are copied it has the wrong values for the schema. I have added the optional parameters and updated the documentation.